llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6552,10 +6552,25 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
     // Note: We call removeProdecessor later since we need to be able to get the
     // PHI value for the default case in case we're using a bit mask.
   } else {
-    Value *Cmp = Builder.CreateICmpULT(
-        TableIndex, ConstantInt::get(MinCaseVal->getType(), TableSize));
-    RangeCheckBranch =
-        Builder.CreateCondBr(Cmp, LookupBB, SI->getDefaultDest());
+    int KeepDefaultBranch = true;
+    if (UseSwitchConditionAsTableIndex) {
+      ConstantRange CR = computeConstantRange(TableIndex, /* ForSigned */ false,
+                                              /* UseInstrInfo*/ true);
+      if (DL.fitsInLegalInteger(CR.getUpper().getZExtValue())) {
```

Comment:
hi @nikic, I think here passing **the value itself** is my intention.
I only want to try this optimization for **small number**, so its table can be fit in a scalar general registrer.
For example, if the switch index is **%x % 15**, then its table size will be 15, rather than 4.

Reply:
Ah yes, you are right.